lib/verifier.rb

```diff
@@ -17,17 +17,23 @@
 class Verifier
 
   class Result
-    attr_accessor :errors, :name
+    attr_accessor :errors, :warnings, :name
     
     def initialize
       @valid = false
+      @safe = false
       @errors = Array.new
+      @warnings = Array.new
     end
     
     def valid?
       @errors.empty?
     end
 
+    def has_warnings?
+      @warnings.empty?
+    end
```

Comment:
The logic is backward here. If there are no warnings (@warnings is empty) the result does not have any warnings and `has_warnings?` should return false.